exporter/signalfxexporter/translation/dpfilters/dimensions.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dpfilters
+
+import (
+	"errors"
+
+	sfxpb "github.com/signalfx/com_signalfx_metrics_protobuf/model"
+)
+
+type dimensionsFilter struct {
+	filterMap map[string]*stringFilter
+}
+
+// newDimensionsFilter returns a filter that matches against a
+// sfxpb.Dimension slice. The filter will return true, if there's
+// at least one matching dimension in the provided slice or if the
```

Comment:
As implemented doesn't dimensionsFilter.Matches return true iff all dimensions match and the slice isn't empty?

Reply:
I believe this branch had an earlier version of the code. Rebased to pick latest commits from PR that was already merged. `dimensionsFilter.Matches` would return `true` if at least one dimension matches and if it's an empty slice, it should return `false` (corrected the comment).

Reply:
I stand corrected. The new version also matches on all dimensions. Will update the comment to reflect that. Thanks for catching!

Reply:
I stand corrected. I've updated the comment to rectify. Thanks for catching!